cwms-data-api/src/main/java/cwms/cda/api/ClobController.java

```diff
@@ -178,13 +187,39 @@ public void getAll(@NotNull Context ctx) {
     public void getOne(@NotNull Context ctx, @NotNull String clobId) {
 
         try (final Timer.Context ignored = markAndTime(GET_ONE)) {
+
+            String idQueryParam = ctx.queryParam(CLOB_ID);
+            if (idQueryParam != null) {
+                clobId = idQueryParam;
+            }
+            String formatHeader = ctx.header(Header.ACCEPT);
+
             DSLContext dsl = getDslContext(ctx);
             ClobDao dao = new ClobDao(dsl);
             Optional<String> office = Optional.ofNullable(ctx.queryParam(OFFICE));
+
+            if ("text/plain".equals(formatHeader)) {
+                // useful cmd:  curl -X 'GET' 'http://localhost:7000/cwms-data/clobs/encoded?office=SPK&id=%2FTIME%20SERIES%20TEXT%2F6261044'
+                // -H 'accept: text/plain' --header "Range: bytes=20000-40000"
+
+                BiConsumer<InputStream, Long> streamConsumer = (stream, length) -> {
+                    if (stream == null) {
+                        ctx.status(HttpServletResponse.SC_NOT_FOUND).json(new CdaError("Unable to find "
+                                + "clob based on given parameters"));
+                    } else {
+                        ctx.seekableStream(stream, "text/plain", length);
+                    }
+                };
+
+                dao.getClob(clobId, office.orElse(null), streamConsumer);
+
+                return;
```

Comment:
Would it be better to put the next section in the "else" instead of returning.

@adamkorynta @ktarbet @DanielTOsborne 

NOTE: purely a question of style.

Reply:
yeah, I'm all for direct simple returns of values in short functions, but total shortcut is just a little jarring.